lib/core/src/model.rs

```diff
@@ -1247,27 +1247,27 @@ impl Payment {
                         refund_tx_amount_sat,
                         ..
                     },
-                ) => Some(PaymentDetails::Lightning {
+                ) => PaymentDetails::Lightning {
                     swap_id,
                     preimage,
                     bolt11,
                     refund_tx_id,
                     refund_tx_amount_sat,
                     description: description.unwrap_or("Liquid transfer".to_string()),
-                }),
+                },
                 Some(PaymentSwapData {
                     swap_type: PaymentSwapType::Chain,
                     swap_id,
                     refund_tx_id,
                     refund_tx_amount_sat,
                     ..
-                }) => Some(PaymentDetails::Bitcoin {
+                }) => PaymentDetails::Bitcoin {
                     swap_id,
                     refund_tx_id,
                     refund_tx_amount_sat,
                     description: description.unwrap_or("Bitcoin transfer".to_string()),
-                }),
-                _ => payment_details,
+                },
+                _ => payment_details.expect("Expected existing payment details"),
```

Comment:
Don't know if `.expect()` is a good idea, as it will panic in case of `None`.

My understanding from the requirements was that it should default to `PaymentDetails::Liquid`.

Since this happens in `from_tx_data`, it means we're trying to reconstruct the details of a pure Liquid tx. Then maybe generating a `PaymentDetails::Liquid` is the way to go?

Reply:
I agree. It seems that the payment_details shouldn't be optional here now after the change.